iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -475,12 +466,23 @@ protected override void Dispose(bool disposing)
             {
                 if (TryStop())
                 {
-                    Cleanup();
+                    CleanUpAsync();
```

Comment:
I know this didn't really change, but doesn't this need to be awaited?

Reply:
This PR is about fixing disposable issues. I'll be addressing unawaited calls in other PRs.